File: auth/src/main/java/com/diagorn/sparkathon/auth/dto/auth/RefreshTokenResponse.java
package com.diagorn.sparkathon.auth.dto.auth;

import lombok.Builder;
import lombok.Data;
import lombok.ToString;

/**
 * Refresh token response
 *
 * @author diagorn
 */
@Data
@Builder
@ToString
public class RefreshTokenResponse {
    /**
     * New access token
     */
    private String accessToken;
    /**
     * New access token expiration timestamp
     */
    private Long expiresIn;
}

File: auth/src/main/java/com/diagorn/sparkathon/auth/dto/auth/RefreshTokenResponse.kt
package com.diagorn.sparkathon.auth.dto.auth

import io.swagger.v3.oas.annotations.media.Schema

/**
 * Refresh token response
 *
 * @author diagorn
 */
class RefreshTokenResponse(
    /**
     * New access token
     */
    @Schema(
        description = "New access token",
        example = "eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJkaWFnb3JuIiwiaWF0IjoxNzU1ODc1MjAzLCJleHAiOjE3NTU4ODM4NDN9.3y8Jbs4Zz0tx7j0u7ZuIbppJASYt1kIl610bMnN7Zuk"
    )
    val accessToken: String,

    /**
     * New access token lifespan (ms)
     */
    @Schema(description = "New access token lifespan (ms)", example = "86400000")
    val expiresIn: Long
) 